src/consensus/aft/raft.h

```diff
@@ -1298,16 +1296,16 @@ namespace aft
       for (auto& ae : append_entries)
       {
         auto& [ds, i] = ae;
-        state->last_idx = i;
```

Comment:
Simplified the logic slightly here: only bump `last_idx` when the entry is successfully applied.